### ISSUE TITLE
Switches not working?

### ISSUE THREAD

**Issue by MortuuS** [ksya/ha-nefiteasy#95]
Hi,

I've been using the Nefit Easy plugin for a while now. Recently I found out all the switches are 'not available' and greyed out.
Therefore I tried reinstalling the plugin through HACS and setting up the device again. Since then it doesn't even add the switches anymore.

Hopefully you can help me resolve this issue. 👍

![Screenshot_2021-12-27-22-40-16-759_io homeassistant companion android](https://user-images.githubusercontent.com/6415298/147580079-8dc1a184-4e98-449f-beb2-b74e8083e0dc.jpg)

The log shows the following error:

> Source: custom_components/nefiteasy/switch.py:64
Integration: Schakelaar (documentation, issues)
First occurred: 22:37:45 (2 occurrences)
Last logged: 22:38:48
Error while setting up nefiteasy platform for switch
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 249, in _async_setup_platform
    await asyncio.shield(task)
  File "/config/custom_components/nefiteasy/switch.py", line 40, in async_setup_entry
    await setup_home_entrance_detection(
  File "/config/custom_components/nefiteasy/switch.py", line 64, in setup_home_entrance_detection
    name=basetypeconf["name"].format(name),
TypeError: string indices must be integers

**Comment by RobBie1221** [comment]
Which home assistant version are you using?

**Comment by MortuuS** [comment]
Hi,
I was running 2021.12.5. But while the checking the version, update 2021.12.6 showed as available. After the update to 2021.12.6. the error in the log remains the same though.
Thanks!

**Comment by RobBie1221** [comment]
Are you using home entrance detection? I see it fails on that. I have to admit, has been a long time since I tested that.

**Comment by MortuuS** [comment]
Yes thats it!

My wife and I have it setup on both our phones. But to be honest I haven't been using it since integrating with HomeAssistant. And the thermostat has been on 'manual' mode since then.

After your suggestion I removed both devices from 'my Nefit account' in the app. Then I restarted HomeAssistant and the error is gone and the switches are back.
For testing purposes I setup presence detection again. The error then comes back.

For now I will turn off presence detection in the app since it is easily replicated in HomeAssistant. I'd rather have the rest of the switches available again!

Thanks again for your help Rob! 👍
